python/hail/table.py

```diff
@@ -2232,32 +2232,39 @@ def explode(self, field, name=None):
             t = t.rename({f: name})
         return t
 
-    @typecheck_method(row_key=expr_any,
-                      col_key=expr_any,
-                      entry_exprs=expr_any)
-    def to_matrix_table(self, row_key, col_key, **entry_exprs):
-
-        from hail.matrixtable import MatrixTable
-
-        all_exprs = []
-        all_exprs.append(row_key)
-        analyze('to_matrix_table/row_key', row_key, self._row_indices)
-        all_exprs.append(col_key)
-        analyze('to_matrix_table/col_key', col_key, self._row_indices)
-
-        exprs = []
-
-        for k, e in entry_exprs.items():
-            all_exprs.append(e)
-            analyze('to_matrix_table/entry_exprs/{}'.format(k), e, self._row_indices)
-            exprs.append('`{k}` = {v}'.format(k=k, v=e._ast.to_hql()))
-
-        base, cleanup = self._process_joins(*all_exprs)
-
-        return MatrixTable(base._jt.toMatrixTable(row_key._ast.to_hql(),
-                                                  col_key._ast.to_hql(),
-                                                  ",\n".join(exprs),
-                                                  joption(None)))
+    @typecheck_method(row_key=sequenceof(str),
+                      col_key=sequenceof(str),
+                      row_fields=sequenceof(str),
+                      col_fields=sequenceof(str),
+                      partition_key=nullable(sequenceof(str)),
+                      n_partitions=nullable(int))
+    def to_matrix_table(self, row_key, col_key, row_fields=[], col_fields=[], partition_key=None, n_partitions=None):
+        all_fields = list(itertools.chain(row_key, col_key, row_fields, col_fields))
+        c = Counter(all_fields)
+        for k, v in c.items():
```

Comment:
Are you eliminating the requirement that the new row and column fields must be row indexed?

Reply:
to_matrix_table was just completely broken in Python. This fixes it. Will add docs / tests after this goes in, if that's OK

Reply:
no, I need to add that check back. good catch